packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3233 @@
+///    separated from other menu items by a divider.
+///  * [MenuBarItem], a leaf menu item which displays the label, an optional
+///    shortcut label, and optional leading and trailing icons.
+///  * [MenuBarController], a class that allows closing of menus from outside of
+///    the menu bar.
+///  * [PlatformMenuBar], which creates a menu bar that is rendered by the host
+///    platform instead of by Flutter (on macOS, for example).
+class MenuBar extends PlatformMenuBar {
+  /// Creates a const [MenuBar].
+  const MenuBar({
+    super.key,
+    this.controller,
+    this.enabled = true,
+    this.backgroundColor,
+    this.height,
+    this.padding,
+    this.elevation,
+    super.menus = const <MenuItem>[],
+  }) : _isPlatformMenu = false;
+
+  // A private constructor for the MenuBar.adaptive factory constructor to use.
+  const MenuBar._({
+    super.key,
+    this.controller,
+    this.enabled = true,
+    this.backgroundColor,
+    this.height,
+    this.padding,
+    this.elevation,
+    bool isPlatformMenu = false,
+    super.menus = const <MenuItem>[],
+  }) : _isPlatformMenu = isPlatformMenu;
+
+  /// Creates an adaptive [MenuBar] that renders using platform APIs with a
+  /// [PlatformMenuBar] on platforms that support it, and using Flutter
+  /// rendering on platforms that don't.
+  factory MenuBar.adaptive({
+    Key? key,
+    MenuBarController? controller,
+    bool enabled = true,
+    MaterialStateProperty<Color?>? backgroundColor,
+    double? height,
+    EdgeInsets? padding,
+    MaterialStateProperty<double?>? elevation,
+    List<MenuItem> menus = const <MenuItem>[],
+  }) {
+    bool isPlatformMenu;
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.iOS:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        isPlatformMenu = false;
+        break;
+      case TargetPlatform.macOS:
+        isPlatformMenu = true;
+        break;
+    }
+
+    return MenuBar._(
+      key: key,
+      controller: controller,
+      enabled: enabled,
+      backgroundColor: backgroundColor,
+      height: height,
+      padding: padding,
+      elevation: elevation,
+      isPlatformMenu: isPlatformMenu,
+      menus: menus,
+    );
+  }
+
+  /// An optional controller that allows outside control of the menu bar.
+  ///
+  /// Setting this controller will allow you to close any open menus from
+  /// outside of the menu bar using [MenuBarController.closeAll].
+  ///
+  /// If a controller is not set, the widget will create its own controller
+  /// internally.
+  final MenuBarController? controller;
+
+  /// Whether or not this menu bar is enabled.
+  ///
+  /// When disabled, all menus are closed, the menu bar buttons are disabled,
+  /// and menu shortcuts are ignored.
+  final bool enabled;
+
+  /// The background color of the menu bar.
+  ///
+  /// Defaults to [MenuBarThemeData.barBackgroundColor] if not set.
```

Comment:
Defaults should be defined and documented by component widgets, not by themes. The default value for all theme properties and for all component widget properties should be null and the implementation should compute the default as needed with something like:
```dart
Foo effectiveFoo = widget.foo ?? theme.foo ?? _defaultTheme.foo!;
```

Reply:
OK. That's how they are implemented. I'll adjust the documentation to show the values from the default theme.

How have we worded that in the past?  For example, is it something like this? "If unset, uses the [MenuBarTheme] value, if that is unset, defaults to XX" or do we just say "If unset, defaults to XX"?

Reply:
We always use "If null, uses the [MenuBarTheme] value, if that is null, defaults to XX".  Saying "unset" isn't quite right. It gets a little more complex for state dependent properties (MaterialStateProperty values) because it's also necessary to explain what states the property is resolved against.